[Feat]#3-MainMenu-JasonYang

VesselWheel opened the pull request:
## Contents
1. Global - Model - 데이터 구조체 선언
2. Drink, Food, Merchandiser 프로퍼티 선언
3. CollectionView 활용 화면구성
4. assets에 이미지파일 추가

## Screenshot
<img width="390" alt="스크린샷 2023-12-27 오후 9 40 23" src="https://github.com/NBCAMP-kiosk-team3/iOS/assets/88497383/98dad2d6-10f4-4ad3-b6a7-c47a642dd710">




## To Reviewers
CollectionView에 Array로 만든 프로퍼티가 들어갈 수 있도록 하겠습니다. 


## Issues
<!-- 이슈가 발생하는 부분이 있다면 적어주세요 -->
CollectionView에 Array로 만든 프로퍼티가 들어가지 않아요. 
addArrangedSubview(_:) 활용해서 수정해보겠습니다.
https://developer.apple.com/documentation/uikit/uistackview/1616227-addarrangedsubview

yenny42 committed:
[Feat] 주문내역 UI 뼈대

yenny42 committed:
[Del] 주석 삭제

yenny42 committed:
[Chore]
- 취소,결제버튼 클릭 시 동작 추가
- Stackview의 서브뷰 정렬 변경
- "직원호출" 버튼 추가
- Tableview 배경색 추가

VesselWheel committed:
[Feat]#3-MainMenu-JasonYang

yenny42 commented:
데이터 구조체를 Group에 담아주시면 좋을 것 같아요~

VesselWheel committed:
Merge remote-tracking branch 'refs/remotes/origin/feat/#4-OrderListUI-yen'

joonBaek12 commented on SPABUCKS-Kiosk-iOS/SPABUCKS-Kiosk-iOS/Presentation/Views/MenuView.swift:
```diff
@@ -4,30 +4,161 @@
- (주석은 확인 후 지워주세요!)
-  **/
-
+import Foundation
```
혹시 Foundation을 임포트하신 이유가 있으실까요!

VesselWheel replied:
제 프로젝트 파일에는 없는데, 커밋이 되었네요. Global 폴더에 Model하위 폴더 만들어서 SpabucksMenuItem 구조체를 만들었는데 그건 반영이 안되었어요.

joonBaek12 commented on SPABUCKS-Kiosk-iOS/SPABUCKS-Kiosk-iOS/Presentation/Views/MenuView.swift:
```diff
@@ -4,30 +4,161 @@
+import Foundation
 import UIKit
 
+
```
p4;
 저희 컨벤션에 따라 공백 삭제 부탁드려요!

VesselWheel replied:
알겠습니다.

joonBaek12 commented on SPABUCKS-Kiosk-iOS/SPABUCKS-Kiosk-iOS/Presentation/Views/MenuView.swift:
```diff
@@ -4,30 +4,161 @@
+}
+
+//MARK: - imageView, nameLabel 셀 설정 서브 클래스
+private class MenuCell: UICollectionViewCell {
```
private 접근제한자를 사용하신 특별한 이유가 있으실까요?

VesselWheel replied:
하위 클래스로 이해하고 적용했는데, 적용할 필요가 없을까요?

joonBaek12 replied:
@cantabilano private은 프로퍼티, 메서드 등 클래스 내부의 멤버에만 적용이 가능한걸로 알고 있어요! 하위클래스이고 더 이상 수정이 되지 않을 것이 보장 될 때 쓰실 수 있는 키워드는 final이 있습니다 :)

joonBaek12 reviewed (approved):
성공하셨군요... 너무 고생하셨습니다!!

VesselWheel committed:
1. 셀의 크기, 간격, 여백, 헤더, 푸터뷰 설정
2. UIImageView와 UILabel을 크기 오토레이아웃 사용하지 않고, 크기 설정
3. 하지만, UIImageView와 UILabel에서 UIImageView에 position ambiguous for UIImageView"와 "UILabel에서는 position ambiguous for UILabel" 오류가 발생 : 두 프로퍼티 간 충돌로 예상

VesselWheel commented:
1. 셀의 크기, 간격, 여백, 헤더, 푸터뷰 설정
2. UIImageView와 UILabel을 크기 오토레이아웃 사용하지 않고, 크기 설정
3. 하지만, UIImageView와 UILabel에서 UIImageView에 position ambiguous for UIImageView"와 "UILabel에서는 position ambiguous for UILabel" 오류가 발생 : 두 프로퍼티 간 충돌로 예상

![issue](https://github.com/NBCAMP-kiosk-team3/iOS/assets/88497383/8921074b-f363-4e5a-b65c-d640d629e862)

VesselWheel commented:
> 데이터 구조체를 Group에 담아주시면 좋을 것 같아요~

좋아요! 적용해보겠습니다.
